src/bblocks/data_importers/world_bank/world_bank.py

```diff
@@ -0,0 +1,559 @@
+        "skipBlanks": skip_blanks,
+        "skipAggs": skip_aggs,
+        "economy": entity_code,
+        "time": time,
+        "numericTimeKeys": True,
+        "params": params,
+        **extra,
+    }
+    api_params = {k: v for k, v in api_params.items() if v is not None}
+
+    # fetch data in batches using multithreading
+    batches = _batch(indicators, batch_size)
+    results: list[pd.DataFrame] = []
+
+    with ThreadPoolExecutor(max_workers=thread_num) as executor:
+        futures = [
+            executor.submit(_request_data, {**api_params, "series": batch_indicators})
+            for batch_indicators in batches
+        ]
+
+        for future in as_completed(futures):
+            results.append(future.result())
+
+    if not results:
+        raise DataExtractionError("No data returned from World Bank API.")
+
+    df = pd.concat(results, ignore_index=True)
+
+    if df.empty:
+        raise DataExtractionError("No data returned from World Bank API.")
+
+    DataFrameValidator().validate(
+        df,
+        required_cols=[
+            Fields.year,
+            Fields.indicator_code,
+            Fields.entity_code,
+            Fields.value,
+        ],
+    )
+
+    logger.info("Data fetched successfully from World Bank API.")
+    return df
+
+
+class WorldBank:
+    """Importer for World Bank data.
+
+    The World Bank offers data through its API for various databases, such as the World Development Indicators (WDI).
+    This importer provides functionality to connect to a specified World Bank database, get data for different indicators,
+    and access metadata on indicators and entities.
+
+    Usage example:
+
+    This class connects to a specified World Bank database using its ID. To see the available databases,
+    use the `get_wb_databases` function.
+    >>> print(get_wb_databases())
+
+    Instantiate the WorldBank class, specifying the database ID you want to use:
+    >>> wb_importer = WorldBank(db=2)  # 2 is the ID
+
+    By default, if no database is specified, the World Development Indicators database (id=2) is used.
+    >>> wb_importer = WorldBank()  # uses WDI by default
+
+    To get available indicators in the database:
+    >>> indicators_df = wb_importer.get_available_indicators()
+
+    To get data for specific indicators:
+    >>> data_df = wb_importer.get_data(indicator_code='SP.POP.TOTL') # total population indicator code
+
+    Multiple indicators can be queried. Batching and multithreading are used to speed up data retrieval for
+    multiple indicators. Indicators are batched into groups of 1 by default, and 4 threads are used for fetching data.
+    Different batch sizes and thread numbers can be specified using the `batch_size` and `thread_num` parameters.
+    Additional parameters can be used including specifying entity codes, year ranges, whether to skip blank
+    observations, whether to include labels, etc. and any other parameter supported by the World Bank API (read the wbgapi
```

Comment:
is this meant to be a markdown link? If so it should be the other way around with the () and []